<div class="alert alert-block alert-info" style="margin-top: 20px">
    <a href="https://cocl.us/corsera_da0101en_notebook_top">
         <img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DA0101EN/Images/TopAd.png" width="750" align="center">
    </a>
</div>


<a href="https://www.bigdatauniversity.com"><img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DA0101EN/Images/CCLog.png" width=300, align="center"></a>

<h1 align=center><font size = 5>Data Analysis with Python</font></h1>

<h1>Introduction</h1>
<h3>Bienvenue!</h3>

<p>
Dans cette section, vous apprendrez à aborder l'acquisition de données de différentes manières et à obtenir les informations nécessaires à partir d'un ensemble de données.
À la fin de cet atelier, vous chargerez avec succès les données dans Jupyter Notebook et obtiendrez des informations fondamentales via la bibliothèque Pandas.</p>

<h2>Table des Matières</h2>

<div class="alert alert-block alert-info" style="margin-top: 20px">
<ol>
    <li><a href="#Acquisition des données">Acquisition des données</a>
    <li><a href="#Notions Basiques">Notions Basiques</a></li>
</ol>

<h1 id="data_acquisition">Data Acquisition</h1>
<p>
Il existe différents formats pour un dataset, .csv, .json, .xlsx, etc. L'ensemble de données peut être stocké à différents endroits, sur votre ordinateur local ou parfois en ligne. <br>
Dans cette section, vous apprendrez à charger un ensemble de données dans notre bloc-notes Jupyter. <br>
Dans notre cas, l'ensemble de données automobile est une source en ligne et est au format CSV (valeurs séparées par des virgules). Utilisons cet ensemble de données comme exemple pour pratiquer la lecture de données.
<ul>
    <li>Source du data: <a href="https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data" target="_blank">https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data</a></li>
    <li>Type du data: csv</li>
</ul>
La bibliothèque Pandas est un outil utile qui nous permet de lire divers ensembles de données dans un bloc de données; nos plates-formes de notebooks Jupyter ont une <b> bibliothèque Pandas </b> intégrée de sorte que tout ce que nous devons faire est d'importer des pandas sans installer.
</p>

In [1]:
# import pandas library
import pandas as pd

<h2> Lire les données </h2>
<p>
Nous utilisons la fonction <code> pandas.read_csv () </code> pour lire le fichier csv. Dans le crochet, nous mettons le chemin du fichier avec un guillemet, de sorte que les pandas liront le fichier dans une trame de données à partir de cette adresse. Le chemin du fichier peut être une URL ou votre adresse de fichier local. <br>
Étant donné que les données n'incluent pas les en-têtes, nous pouvons ajouter un argument <code> headers = None </code> dans la méthode <code> read_csv () </code>, afin que les pandas ne définissent pas automatiquement la première ligne comme en-tête . <br>
Vous pouvez également affecter l'ensemble de données à n'importe quelle variable que vous créez.
</p>

In [2]:
# Import pandas library
import pandas as pd

# Lecture du fichier via une URL et son attribution à une variable "df"
#other_path = "https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DA0101EN/auto.csv"

# Lecture du fichier depuis le disque local et son attribution à une variable "df"
#other_path = r"C:\PYTHON PRACTICE\2_Data Science project\1_Importation des data/auto.csv"
df = pd.read_csv(r"C:\python_pratique\data science\1_Importation des data/auto.csv", header=None)
#df = pd.read_csv(other_path, header=None)

In [3]:
print(df)

     0    1            2       3      4     5            6    7      8   \
0     3    ?  alfa-romero     gas    std   two  convertible  rwd  front   
1     3    ?  alfa-romero     gas    std   two  convertible  rwd  front   
2     1    ?  alfa-romero     gas    std   two    hatchback  rwd  front   
3     2  164         audi     gas    std  four        sedan  fwd  front   
4     2  164         audi     gas    std  four        sedan  4wd  front   
..   ..  ...          ...     ...    ...   ...          ...  ...    ...   
200  -1   95        volvo     gas    std  four        sedan  rwd  front   
201  -1   95        volvo     gas  turbo  four        sedan  rwd  front   
202  -1   95        volvo     gas    std  four        sedan  rwd  front   
203  -1   95        volvo  diesel  turbo  four        sedan  rwd  front   
204  -1   95        volvo     gas  turbo  four        sedan  rwd  front   

        9   ...   16    17    18    19    20   21    22  23  24     25  
0     88.6  ...  130  mpfi

Après avoir lu le dataset, nous pouvons utiliser la méthode <code> dataframe.head (n) </code> pour vérifier les n premières lignes du dataframe; où n est un entier. Contrairement à <code> dataframe.head (n) </code>, <code> dataframe.tail (n) </code> vous montrera les n dernières lignes du dataframe.

In [4]:
# show the first 5 rows using dataframe.head() method
print("Les 5 premières lignes du dataset_dataframe") 
df.head(5)

Les 5 premières lignes du dataset_dataframe


0    1            2    3    4     5            6    7      8     9   ...  \
0   3    ?  alfa-romero  gas  std   two  convertible  rwd  front  88.6  ...   
1   3    ?  alfa-romero  gas  std   two  convertible  rwd  front  88.6  ...   
2   1    ?  alfa-romero  gas  std   two    hatchback  rwd  front  94.5  ...   
3   2  164         audi  gas  std  four        sedan  fwd  front  99.8  ...   
4   2  164         audi  gas  std  four        sedan  4wd  front  99.4  ...   

    16    17    18    19    20   21    22  23  24     25  
0  130  mpfi  3.47  2.68   9.0  111  5000  21  27  13495  
1  130  mpfi  3.47  2.68   9.0  111  5000  21  27  16500  
2  152  mpfi  2.68  3.47   9.0  154  5000  19  26  16500  
3  109  mpfi  3.19  3.40  10.0  102  5500  24  30  13950  
4  136  mpfi  3.19  3.40   8.0  115  5500  18  22  17450  

[5 rows x 26 columns]

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #1: </h1>
<b>Afficher les dernières lignes du dataset "df".</b>
</div>

In [5]:
# Write your code below and press Shift+Enter to execute 
print("Les 10 dernières lignes\n")


Les 10 dernières lignes



<h3> Ajouter des en-têtes </h3>
<p>
Jetez un œil à notre ensemble de données; pandas définit automatiquement l'en-tête par un entier à partir de 0.
</p>
<p>
Pour mieux décrire nos données, nous pouvons introduire un en-tête, ces informations sont disponibles à l'adresse: <a href="https://archive.ics.uci.edu/ml/datasets/Automobile" target="_blank"> https: // archive.ics.uci.edu/ml/datasets/Automobile </a>
</p>
<p>
Ainsi, nous devons ajouter des en-têtes manuellement.
</p>
<p>
Tout d'abord, nous créons une liste "en-têtes" qui inclut tous les noms de colonnes dans l'ordre.
Ensuite, nous utilisons <code> dataframe.columns = headers </code> pour remplacer les en-têtes par la liste que nous avons créée.
</p>

In [6]:
# create headers list
headers = ["symboling","normalized-losses","make","fuel-type","aspiration", "num-of-doors","body-style",
         "drive-wheels","engine-location","wheel-base", "length","width","height","curb-weight","engine-type",
         "num-of-cylinders", "engine-size","fuel-system","bore","stroke","compression-ratio","horsepower",
         "peak-rpm","city-mpg","highway-mpg","price"]
print("headers\n", headers)

headers
 ['symboling', 'normalized-losses', 'make', 'fuel-type', 'aspiration', 'num-of-doors', 'body-style', 'drive-wheels', 'engine-location', 'wheel-base', 'length', 'width', 'height', 'curb-weight', 'engine-type', 'num-of-cylinders', 'engine-size', 'fuel-system', 'bore', 'stroke', 'compression-ratio', 'horsepower', 'peak-rpm', 'city-mpg', 'highway-mpg', 'price']


Nous remplaçons les en-têtes et revérifions notre dataset

In [7]:
df.columns = headers
df.head(10)

symboling normalized-losses         make fuel-type aspiration num-of-doors  \
0          3                 ?  alfa-romero       gas        std          two   
1          3                 ?  alfa-romero       gas        std          two   
2          1                 ?  alfa-romero       gas        std          two   
3          2               164         audi       gas        std         four   
4          2               164         audi       gas        std         four   
5          2                 ?         audi       gas        std          two   
6          1               158         audi       gas        std         four   
7          1                 ?         audi       gas        std         four   
8          1               158         audi       gas      turbo         four   
9          0                 ?         audi       gas      turbo          two   

    body-style drive-wheels engine-location  wheel-base  ...  engine-size  \
0  convertible          rwd           front        88.6  ...          130   
1  convertible          rwd           front        88.6  ...          130   
2    hatchback          rwd           front        94.5  ...          152   
3        sedan          fwd           front        99.8  ...          109   
4        sedan          4wd           front        99.4  ...          136   
5        sedan          fwd           front        99.8  ...          136   
6        sedan          fwd           front       105.8  ...          136   
7        wagon          fwd           front       105.8  ...          136   
8        sedan          fwd           front       105.8  ...          131   
9    hatchback          4wd           front        99.5  ...          131   

   fuel-system  bore  stroke compression-ratio horsepower  peak-rpm city-mpg  \
0         mpfi  3.47    2.68               9.0        111      5000       21   
1         mpfi  3.47    2.68               9.0        111      5000       21   
2         mpfi  2.68    3.47               9.0        154      5000       19   
3         mpfi  3.19    3.40              10.0        102      5500       24   
4         mpfi  3.19    3.40               8.0        115      5500       18   
5         mpfi  3.19    3.40               8.5        110      5500       19   
6         mpfi  3.19    3.40               8.5        110      5500       19   
7         mpfi  3.19    3.40               8.5        110      5500       19   
8         mpfi  3.13    3.40               8.3        140      5500       17   
9         mpfi  3.13    3.40               7.0        160      5500       16   

  highway-mpg  price  
0          27  13495  
1          27  16500  
2          26  16500  
3          30  13950  
4          22  17450  
5          25  15250  
6          25  17710  
7          25  18920  
8          20  23875  
9          22      ?  

[10 rows x 26 columns]

nous pouvons supprimer les valeurs manquantes dans la colonne "prix" comme suit

In [7]:
df.dropna(subset=["price"], axis=0)

symboling normalized-losses         make fuel-type aspiration  \
0            3                 ?  alfa-romero       gas        std   
1            3                 ?  alfa-romero       gas        std   
2            1                 ?  alfa-romero       gas        std   
3            2               164         audi       gas        std   
4            2               164         audi       gas        std   
..         ...               ...          ...       ...        ...   
200         -1                95        volvo       gas        std   
201         -1                95        volvo       gas      turbo   
202         -1                95        volvo       gas        std   
203         -1                95        volvo    diesel      turbo   
204         -1                95        volvo       gas      turbo   

    num-of-doors   body-style drive-wheels engine-location  wheel-base  ...  \
0            two  convertible          rwd           front        88.6  ...   
1            two  convertible          rwd           front        88.6  ...   
2            two    hatchback          rwd           front        94.5  ...   
3           four        sedan          fwd           front        99.8  ...   
4           four        sedan          4wd           front        99.4  ...   
..           ...          ...          ...             ...         ...  ...   
200         four        sedan          rwd           front       109.1  ...   
201         four        sedan          rwd           front       109.1  ...   
202         four        sedan          rwd           front       109.1  ...   
203         four        sedan          rwd           front       109.1  ...   
204         four        sedan          rwd           front       109.1  ...   

     engine-size  fuel-system  bore  stroke compression-ratio horsepower  \
0            130         mpfi  3.47    2.68               9.0        111   
1            130         mpfi  3.47    2.68               9.0        111   
2            152         mpfi  2.68    3.47               9.0        154   
3            109         mpfi  3.19    3.40              10.0        102   
4            136         mpfi  3.19    3.40               8.0        115   
..           ...          ...   ...     ...               ...        ...   
200          141         mpfi  3.78    3.15               9.5        114   
201          141         mpfi  3.78    3.15               8.7        160   
202          173         mpfi  3.58    2.87               8.8        134   
203          145          idi  3.01    3.40              23.0        106   
204          141         mpfi  3.78    3.15               9.5        114   

     peak-rpm city-mpg highway-mpg  price  
0        5000       21          27  13495  
1        5000       21          27  16500  
2        5000       19          26  16500  
3        5500       24          30  13950  
4        5500       18          22  17450  
..        ...      ...         ...    ...  
200      5400       23          28  16845  
201      5300       19          25  19045  
202      5500       18          23  21485  
203      4800       26          27  22470  
204      5400       19          25  22625  

[205 rows x 26 columns]

Maintenant, nous avons lu avec succès l'ensemble de données brutes et ajouté les en-têtes corrects dans le bloc de données.

 <div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #2: </h1>
<b>Trouvez le nom des colonnes du dataframe</b>
</div>

In [8]:
# Ecrit votre code en bas et faire shift+Entrer 


<h2> Enregistrer l'ensemble de données </h2>
<p>
En conséquence, Pandas nous permet d'enregistrer l'ensemble de données au format csv en utilisant la méthode <code> dataframe.to_csv () </code>, vous pouvez ajouter le chemin et le nom du fichier avec des guillemets entre crochets.
</p>
<p>
     Par exemple, si vous enregistrez le dataframe <b> df </b> sous <b> automobile.csv </b> sur votre ordinateur local, vous pouvez utiliser la syntaxe ci-dessous:
</p>

Nous pouvons également lire et enregistrer d'autres formats de fichiers, nous pouvons utiliser des fonctions similaires à ** `pd.read_csv ()` ** et ** `df.to_csv ()` ** pour d'autres formats de données, les fonctions sont répertoriées dans le tableau suivant:

<h2>Read/Save Other Data Formats</h2>



| Data Formate  | Read           | Save             |
| ------------- |:--------------:| ----------------:|
| csv           | `pd.read_csv()`  |`df.to_csv()`     |
| json          | `pd.read_json()` |`df.to_json()`    |
| excel         | `pd.read_excel()`|`df.to_excel()`   |
| hdf           | `pd.read_hdf()`  |`df.to_hdf()`     |
| sql           | `pd.read_sql()`  |`df.to_sql()`     |
| ...           |   ...          |       ...        |

<h1 id = "basic_insight"> Notions de bases </h1>
<p>
Après avoir lu les données dans le cadre de données Pandas, il est temps pour nous d'explorer l'ensemble de données. <br>
Il existe plusieurs façons d'obtenir des informations essentielles sur les données afin de nous aider à mieux comprendre notre ensemble de données.
</p>

<h2> Types de données </h2>
<p>
Les données sont de différents types. <br>
Les principaux types stockés dans les dataframes Pandas sont <b> object </b>, <b> float </b>, <b> int </b>, <b> bool </b> et <b> datetime64 </ b>. Afin de mieux connaître chaque attribut, il est toujours bon pour nous de connaître le type de données de chaque colonne. Chez les pandas:
</p>

In [9]:
df.dtypes


symboling              int64
normalized-losses     object
make                  object
fuel-type             object
aspiration            object
num-of-doors          object
body-style            object
drive-wheels          object
engine-location       object
wheel-base           float64
length               float64
width                float64
height               float64
curb-weight            int64
engine-type           object
num-of-cylinders      object
engine-size            int64
fuel-system           object
bore                  object
stroke                object
compression-ratio    float64
horsepower            object
peak-rpm              object
city-mpg               int64
highway-mpg            int64
price                 object
dtype: object

In [10]:
# vérifier le type de données de la trame de données "df" avec .dtypes
print(df.dtypes)

symboling              int64
normalized-losses     object
make                  object
fuel-type             object
aspiration            object
num-of-doors          object
body-style            object
drive-wheels          object
engine-location       object
wheel-base           float64
length               float64
width                float64
height               float64
curb-weight            int64
engine-type           object
num-of-cylinders      object
engine-size            int64
fuel-system           object
bore                  object
stroke                object
compression-ratio    float64
horsepower            object
peak-rpm              object
city-mpg               int64
highway-mpg            int64
price                 object
dtype: object


<p>
En conséquence, comme indiqué ci-dessus, il est clair que le type de données "symboling" et "curb-weight" est <code> int64 </code>, "normalized-loss" est <code> object </ code >, et "wheel-base" est <code> float64 </code>, etc.
</p>
<p>
Ces types de données peuvent être modifiés; nous apprendrons comment y parvenir dans un module ultérieur.
</p>

<h2> Description des colonnes </h2>
Si nous souhaitons obtenir un résumé statistique de chaque colonne, tel que le nombre, la valeur moyenne de la colonne, l'écart type de la colonne, etc. Nous utilisons la méthode describe:

Cette méthode fournira diverses statistiques récapitulatives, à l'exclusion des valeurs <code> NaN </code> (Not a Number).

In [11]:
df.describe()

symboling  wheel-base      length       width      height  \
count  205.000000  205.000000  205.000000  205.000000  205.000000   
mean     0.834146   98.756585  174.049268   65.907805   53.724878   
std      1.245307    6.021776   12.337289    2.145204    2.443522   
min     -2.000000   86.600000  141.100000   60.300000   47.800000   
25%      0.000000   94.500000  166.300000   64.100000   52.000000   
50%      1.000000   97.000000  173.200000   65.500000   54.100000   
75%      2.000000  102.400000  183.100000   66.900000   55.500000   
max      3.000000  120.900000  208.100000   72.300000   59.800000   

       curb-weight  engine-size  compression-ratio    city-mpg  highway-mpg  
count   205.000000   205.000000         205.000000  205.000000   205.000000  
mean   2555.565854   126.907317          10.142537   25.219512    30.751220  
std     520.680204    41.642693           3.972040    6.542142     6.886443  
min    1488.000000    61.000000           7.000000   13.000000    16.000000  
25%    2145.000000    97.000000           8.600000   19.000000    25.000000  
50%    2414.000000   120.000000           9.000000   24.000000    30.000000  
75%    2935.000000   141.000000           9.400000   30.000000    34.000000  
max    4066.000000   326.000000          23.000000   49.000000    54.000000

<p>
Ceci montre le résumé statistique de toutes les colonnes de type numérique (int, float). <br>
Par exemple, l'attribut «symboling» a 205 points, la valeur moyenne de cette colonne est 0,83, l'écart type est 1,25, la valeur minimale est -2, le 25e centile est 0, le 50e centile est 1, le 75e centile est 2 et la valeur maximale est 3.
<br>
Cependant, que se passe-t-il si nous souhaitons également vérifier toutes les colonnes, y compris celles de type objet.
<br> <br>

Vous pouvez ajouter un argument <code> include = "all" </code> à l'intérieur du crochet. Essayons à nouveau.
</p> 

In [12]:
# describe all the columns in "df" 
df.describe(include = "all")

symboling normalized-losses    make fuel-type aspiration  \
count   205.000000               205     205       205        205   
unique         NaN                52      22         2          2   
top            NaN                 ?  toyota       gas        std   
freq           NaN                41      32       185        168   
mean      0.834146               NaN     NaN       NaN        NaN   
std       1.245307               NaN     NaN       NaN        NaN   
min      -2.000000               NaN     NaN       NaN        NaN   
25%       0.000000               NaN     NaN       NaN        NaN   
50%       1.000000               NaN     NaN       NaN        NaN   
75%       2.000000               NaN     NaN       NaN        NaN   
max       3.000000               NaN     NaN       NaN        NaN   

       num-of-doors body-style drive-wheels engine-location  wheel-base  ...  \
count           205        205          205             205  205.000000  ...   
unique            3          5            3               2         NaN  ...   
top            four      sedan          fwd           front         NaN  ...   
freq            114         96          120             202         NaN  ...   
mean            NaN        NaN          NaN             NaN   98.756585  ...   
std             NaN        NaN          NaN             NaN    6.021776  ...   
min             NaN        NaN          NaN             NaN   86.600000  ...   
25%             NaN        NaN          NaN             NaN   94.500000  ...   
50%             NaN        NaN          NaN             NaN   97.000000  ...   
75%             NaN        NaN          NaN             NaN  102.400000  ...   
max             NaN        NaN          NaN             NaN  120.900000  ...   

        engine-size  fuel-system  bore  stroke compression-ratio horsepower  \
count    205.000000          205   205     205        205.000000        205   
unique          NaN            8    39      37               NaN         60   
top             NaN         mpfi  3.62    3.40               NaN         68   
freq            NaN           94    23      20               NaN         19   
mean     126.907317          NaN   NaN     NaN         10.142537        NaN   
std       41.642693          NaN   NaN     NaN          3.972040        NaN   
min       61.000000          NaN   NaN     NaN          7.000000        NaN   
25%       97.000000          NaN   NaN     NaN          8.600000        NaN   
50%      120.000000          NaN   NaN     NaN          9.000000        NaN   
75%      141.000000          NaN   NaN     NaN          9.400000        NaN   
max      326.000000          NaN   NaN     NaN         23.000000        NaN   

        peak-rpm    city-mpg highway-mpg price  
count        205  205.000000  205.000000   205  
unique        24         NaN         NaN   187  
top         5500         NaN         NaN     ?  
freq          37         NaN         NaN     4  
mean         NaN   25.219512   30.751220   NaN  
std          NaN    6.542142    6.886443   NaN  
min          NaN   13.000000   16.000000   NaN  
25%          NaN   19.000000   25.000000   NaN  
50%          NaN   24.000000   30.000000   NaN  
75%          NaN   30.000000   34.000000   NaN  
max          NaN   49.000000   54.000000   NaN  

[11 rows x 26 columns]

<p>
Désormais, il fournit le résumé statistique de toutes les colonnes, y compris les attributs de type objet. <br>
Nous pouvons maintenant voir le nombre de valeurs uniques, qui est la valeur la plus élevée et la fréquence de la valeur la plus élevée dans les colonnes de type objet. <br>
Certaines valeurs du tableau ci-dessus s'affichent sous la forme "NaN", car ces chiffres ne sont pas disponibles pour un type de colonne particulier. <br>
</p>

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Question #3: </h1>

<p>
Vous pouvez sélectionner les colonnes d'un bloc de données en indiquant le nom de chaque colonne, par exemple, vous pouvez sélectionner les trois colonnes comme suit:
</p>
<p>
     <code> dataframe [['colonne 1', colonne 2 ',' colonne 3 ']] </code>
</p>
<p>
Où "column" est le nom de la colonne, vous pouvez appliquer la méthode ".describe ()" pour obtenir les statistiques de ces colonnes comme suit:
</p>
<p>
     <code> dataframe [['colonne 1', colonne 2 ',' colonne 3 ']] .describe () </code>
</p>

Appliquez la méthode ".describe ()" aux colonnes 'length' et 'compression-ratio'.
</div>

In [13]:
# Write your code below and press Shift+Enter to execute 


<h2>Info</h2>
Another method you can use to check your dataset is:

It provide a concise summary of your DataFrame.

In [14]:
# look at the info of "df"
df.info

<bound method DataFrame.info of      symboling normalized-losses         make fuel-type aspiration  \
0            3                 ?  alfa-romero       gas        std   
1            3                 ?  alfa-romero       gas        std   
2            1                 ?  alfa-romero       gas        std   
3            2               164         audi       gas        std   
4            2               164         audi       gas        std   
..         ...               ...          ...       ...        ...   
200         -1                95        volvo       gas        std   
201         -1                95        volvo       gas      turbo   
202         -1                95        volvo       gas        std   
203         -1                95        volvo    diesel      turbo   
204         -1                95        volvo       gas      turbo   

    num-of-doors   body-style drive-wheels engine-location  wheel-base  ...  \
0            two  convertible          rwd      

<p>
Ici, nous pouvons voir les informations de notre dataframe, avec les 30 premières lignes et les 30 dernières lignes.
<br> <br>
Et, il nous montre également que la trame de données entière a 205 lignes et 26 colonnes au total.
</p>

<h1> Excellent! Vous venez de terminer l'introduction sur l'importation des données! </h1>

<div class="alert alert-block alert-info" style="margin-top: 20px">

    <p><a href="https://cocl.us/corsera_da0101en_notebook_bottom"><img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DA0101EN/Images/BottomAd.png" width="750" align="center"></a></p>
</div>


<h3>About the Authors:</h3>

This notebook was written by <a href="https://www.linkedin.com/in/mahdi-noorian-58219234/" target="_blank">Mahdi Noorian PhD</a>, <a href="https://www.linkedin.com/in/joseph-s-50398b136/" target="_blank">Joseph Santarcangelo</a>, Bahare Talayian, Eric Xiao, Steven Dong, Parizad, Hima Vsudevan and <a href="https://www.linkedin.com/in/fiorellawever/" target="_blank">Fiorella Wenver</a> and <a href=" https://www.linkedin.com/in/yi-leng-yao-84451275/ " target="_blank" >Yi Yao</a>.

<p><a href="https://www.linkedin.com/in/joseph-s-50398b136/" target="_blank">Joseph Santarcangelo</a> is a Data Scientist at IBM, and holds a PhD in Electrical Engineering. His research focused on using Machine Learning, Signal Processing, and Computer Vision to determine how videos impact human cognition. Joseph has been working for IBM since he completed his PhD.</p>

<hr>
<p>Copyright &copy; 2018 IBM Developer Skills Network. This notebook and its source code are released under the terms of the <a href="https://cognitiveclass.ai/mit-license/">MIT License</a>.</p>